projects/plugins/jetpack/_inc/client/state/recommendations/reducer.js

```diff
@@ -467,38 +566,112 @@ const isStepEligibleToShow = ( state, step ) => {
 			return isConditionalRecommendationEnabled( state, step ) && ! isFeatureActive( state, step );
 		case 'boost':
 			return isConditionalRecommendationEnabled( state, step ) && ! isFeatureActive( state, step );
+		case 'welcome__complete':
+		case 'welcome__security':
+		case 'welcome__antispam':
+		case 'welcome__videopress':
+		case 'welcome__search':
+		case 'welcome__scan':
+		case 'backup-activated':
+		case 'scan-activated':
+		case 'antispam-activated':
+		case 'videopress-activated':
+		case 'search-activated':
+		case 'server-credentials':
+			return true; // Can we check if credentials were added?
+		case 'welcome__backup':
+			return getStepsForOnboarding( state ).includes( 'welcome__backup' );
 		default:
 			return ! isFeatureActive( state, step );
 	}
 };
 
 const getNextEligibleStep = ( state, step ) => {
-	let nextStep = stepToNextStep[ step ];
+	const active = get( getOnboardingData( state ), 'active' );
+	const stepToNextStep = get( stepToNextStepByPath, active ? `onboarding.${ active }` : 'default' );
+
+	if ( ! stepToNextStep ) {
+		throw 'Next eligible steps: No path found!';
```

Comment:
I wonder if we could send them to the summary page in this scenario instead of throwing an error. Would probably be a better user experience if something were to happen that triggered this